### PR TITLE
CustomQueue

### PR DESCRIPTION
**Pull request by Pavel-Lapatin**
Task in the directory  Queue Task

### CODE REVIEW

File: lab04/Queue Task/QueueDemonstrate/CustomQueue.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Threading.Tasks;
+
+namespace QueueDemonstrate
+{
+    public class CustomQueue<T> : IQueue<T> where T : class
+    {
+        private QueueElement<T> FirstElemetn;
+        private QueueElement<T> LastElement;
+        private QueueElement<T> CurrentElement { get; set; }
+
```
Comment (pryabov) [inline lab04/Queue Task/QueueDemonstrate/CustomQueue.cs]: one extra line can be removed

File: lab04/Queue Task/QueueDemonstrate/CustomQueue.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using System.Linq;
+using System.Security.AccessControl;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QueueDemonstrate
+{
+    public class CustomQueue<T> : IQueue<T> where T : class
+    {
+        private QueueElement<T> FirstElemetn;
```
Comment (pryabov) [inline lab04/Queue Task/QueueDemonstrate/CustomQueue.cs]: `FirstElemetn` -> `FirstElement`

File: lab04/Queue Task/QueueDemonstrate/QueueElement.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QueueDemonstrate
+{
+    public class QueueElement<T> where T : class
+    {
+        public T Current { get; set; }
```
Comment (pryabov) [inline lab04/Queue Task/QueueDemonstrate/QueueElement.cs]: `Current` better to rename to `Value` or something else.

File: lab04/Queue Task/QueueDemonstrate/QueueElement.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        public QueueElement<T> Parent { get; set; }
+        public QueueElement<T> Previous { get; set; }
```
Comment (pryabov) [inline lab04/Queue Task/QueueDemonstrate/QueueElement.cs]: Do we need `Previous`?

File: lab04/task03.sql [hunk 5]
```diff
@@ -1,136 +0,0 @@
-﻿--Task 3
```
Comment (pryabov) [inline lab04/task03.sql]: Why was this file removed?

**Comment by pryabov** [comment]
Pls, fix few brief comments and merge to master

**Comment by Pavel-Lapatin** [comment]
Hi. Do you need to approve this commit or I could merge it?
